### DISCUSSION TITLE
Storing the MapControls position/rotation and then setting

### DISCUSSION THREAD

**Discussion by bknill** [pmndrs/drei#884] [Q&A] [answered]
I'm attempting to store the saved position, rotation of camera when the controls onEnd event is fired

```
  const onEnd = () => {
    dispatch(
      sceneEditorActions.updateCameraState({
        position: { id: offlineId, offlineId, x: camera.position.x, y: camera.position.y, z: camera.position.z },
        rotation: { id: offlineId, offlineId, x: camera.rotation.x, y: camera.rotation.y, z: camera.rotation.z },
      }),
    );
  }
```

i.e. the camera is in this position zoomed in on this image
![image](https://user-images.githubusercontent.com/7074483/166174515-bf8328c5-7072-4695-a4ba-e8048cb3f8a4.png)

And then set this camera position when the scene loads 

```
  useEffect(() => {
    if (!settings) return;
    if (settings.position) camera.position.set(settings.position.x, settings.position.y, settings.position.z);
   if (settings.rotation) camera.rotation.set(settings.rotation.x, settings.rotation.y, settings.rotation.z);
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [ settings]);
```
![image](https://user-images.githubusercontent.com/7074483/166174588-bfbd3227-0b82-4135-ae18-b3dbe1e827c5.png)

But as you can see the camera or plane is on a strange angle. 

Whats the best way to store the Controls settings (I tried with target as well with same outcome) and then restoring them? 

I'm also gonna want to animate these next.

**Comment by bknill** [comment, marked as answer]
Ahhh worked it out. 

Saving camera position and controls target and then applying those worked.

**Reply by mwmwmw** [reply]
Ha, I was just posting this!

**Reply by mwmwmw** [reply]
I have seen this rotation issue before with MapControls. I can't be sure the issue is related to what you are doing.  But it's worth a shot. 

You can see that discussion here https://github.com/pmndrs/react-three-fiber/discussions/2205.

To fix it we realized that the camera and a ref of map controls position needed to be set.

I can't say for sure this will solve your problem, but it might be a good starting place. 


-- Tinker around with this to reproduce the issue. 

Here is the codesandbox used to solve the issue.
https://codesandbox.io/s/mapcontrols-constrained-059y7o

Try commenting out just the camera, on line 47 and 51. Pan around and you'll see the rotation at the boundary. 
You can achieve a slight variation of the glitch by commenting out 48 and 52,

**Reply by bknill** [reply]
Thanks dude! Yea this is working great
